### ISSUE TITLE
Segmentation fault with MH01 and MH02 Euroc Scenes

### ISSUE THREAD

**Issue by AlbertoRemus** [zzzzxxxx111/Rover-SLAM#6]
This happens with Stereo-Inertial setup:

**MH01**:
```
Not IMU data in mlQueueImuData!!
not IMU meas
not IMU meas
Gtk-Message: 12:59:53.505: Failed to load module "canberra-gtk-module"
Starting the Viewer
not enough acceleration
not enough acceleration
First KF:0; Map init KF:0
New Map created with 1293 points
Segmentation fault (core dumped)
```

**MH02**:
```
Not IMU data in mlQueueImuData!!
not IMU meas
not IMU meas
Gtk-Message: 12:53:21.528: Failed to load module "canberra-gtk-module"
Starting the Viewer
not enough acceleration
not enough acceleration
not enough acceleration
not enough acceleration
not enough acceleration
not enough acceleration
First KF:0; Map init KF:0
New Map created with 812 points
Segmentation fault (core dumped)

```

All the other scenes are ok.

Do you have any insight about how to debug this issue?

**Comment by t676290894** [comment]
I encountered the same problem, and when I ran monocular MH03 and MH05, I found that the number of matches decreased towards the end. Afterwards, I used EVO tool to evaluate these two trajectories and found that the RMSE was relatively large with poor accuracy. I don't know if it's related to the version. My device environment is CUDA12.1. onnxruntime1.18，opencv4.2

**Comment by t676290894** [comment]
I ran monocular inertial MH05 with an RMSE of 0.065; When running monocular inertial MH03, pVp ->mpcpi is empty, causing program interruption

**Comment by t676290894** [comment]
I ran monocular inertial MH05 and the result was rmse of 0.065; When running monocular inertial MH03, pVp ->mpcpi is empty, causing program interruption

**Comment by AlbertoRemus** [comment]
I've noticed that in the stereo case, it seems it is connected to a division by 0
https://github.com/zzzzxxxx111/Rover-SLAM/blob/452a37a6d4950762c13c5a9472bb3708302fb3a1/src/LocalMapping.cc#L947-L952

That was not present in the original ORBSLAM3 repo

https://github.com/UZ-SLAMLab/ORB_SLAM3/blob/4452a3c4ab75b1cde34e5505a36ec3f9edcdc4c4/src/LocalMapping.cc#L712

Since I've noticed that `lastmatchtrack` and `lastmatchnum` are not used, removing those, solved the issue.

**Comment by t676290894** [comment]
Thank you for your help. I have re run the MH03 and MH05 datasets with  stereo -inertia, and RMSE is indeed consistent with the paper, with good results. I feel that this is closely related to a well functioning closed-loop system.

**Comment by AlbertoRemus** [comment]
Thanks for the feedback @t676290894, glad it helped, since I have a question related to quantitative results I will close this issue to open a new one

**Comment by AlbertoRemus** [comment]
> Thanks for the feedback @t676290894, glad it helped, since I have a question related to quantitative results I will close this issue to open a new one

See issue #7

**Comment by zzzzxxxx111** [comment]
> Thank you for your help. I have re run the MH03 and MH05 datasets with stereo -inertia, and RMSE is indeed consistent with the paper, with good results. I feel that this is closely related to a well functioning closed-loop system.
Thank you for your test, I just saw your issue, if the issue is not solved, please continue to ask me questions

**Comment by zzzzxxxx111** [comment]
> I encountered the same problem, and when I ran monocular MH03 and MH05, I found that the number of matches decreased towards the end. Afterwards, I used EVO tool to evaluate these two trajectories and found that the RMSE was relatively large with poor accuracy. I don't know if it's related to the version. My device environment is CUDA12.1. onnxruntime1.18，opencv4.2

Hello, in monocular mode, the size of the input image is more important, because the deep learning model is a fixed size training set, after testing, Limiting the size of the input image to 350-650 works well.

**Comment by zzzzxxxx111** [comment]
> I encountered the same problem, and when I ran monocular MH03 and MH05, I found that the number of matches decreased towards the end. Afterwards, I used EVO tool to evaluate these two trajectories and found that the RMSE was relatively large with poor accuracy. I don't know if it's related to the version. My device environment is CUDA12.1. onnxruntime1.18，opencv4.2

Hello, in mono and mono-vi mode, the size of the input image is more important, because the deep learning model is a fixed size training set, after testing, Limiting the size of the input image to 350-650 works well.

**Comment by zzzzxxxx111** [comment]
> Thank you for your help. I have re run the MH03 and MH05 datasets with stereo -inertia, and RMSE is indeed consistent with the paper, with good results. I feel that this is closely related to a well functioning closed-loop system.

Thank you for your test, I just saw your issue, if the issue is not solved, please continue to ask me questions
